sdk/src/main/java/io/opentelemetry/sdk/trace/SpanInterceptor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import io.opentelemetry.trace.Span;
+
+/**
+ * SpanInterceptor is the interface {@code TracerSdk} uses to allow synchronous hooks for when a
+ * {@code Span} is started or when a {@code Span} is ended.
+ */
+public interface SpanInterceptor {
+  /**
+   * Called when a {@link io.opentelemetry.trace.Span} is started, if the {@link
+   * Span#isRecordingEvents()} returns true.
+   *
+   * @param span the {@code SpanSdk} that just started.
+   */
+  void onStartSync(SpanSdk span);
+
+  /**
+   * Called when a {@link io.opentelemetry.trace.Span} is ended, if the {@link
```

Comment:
Need to clarify whether additional attributes can be set on `SpanSdk` at this point. Basically mention it happens before the span will be passed to exporters

Reply:
Exporter will be a SpanInterceptor as well. Was thinking that this is called after the span is ended so no ability to add more data.

Reply:
oh, I see. I thought interceptor is a mechanism of callbacks to operate on spans. Like modify tracestate and such. You basically replace Exporter API with interceptor.

Should we just call it Exporter?

Reply:
Oh, I see. Interceptor will call exporters, potentially in batches

Reply:
Exporter will be built on top of this raw API.